File: Easy leetcode/504. Base 7.cpp
class Solution {
public:
    string convertToBase7(int num) {
        if (num == 0) return "0";

        bool isNegative = num < 0;
        num = abs(num);

        string result = "";

        while (num) {
            result = to_string(num % 7) + result;
            num /= 7;
        }

        if (isNegative) result = "-" + result;

        return result;
    }
};

File: Easy leetcode/504. Base 7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"

        is_negative = num < 0
        num = abs(num)

        result = ""

        while num > 0:
            result = str(num % 7) + result
            num //= 7

        if is_negative:
            result = "-" + result

        return result

